crates/astria-composer/tests/blackbox/executor.rs

```diff
@@ -0,0 +1,224 @@
+    let expected_rollup_data_submissions = [seq0];
+    let requests = response_guard.received_requests().await;
+    assert_eq!(requests.len(), 1);
+
+    // verify the expected sequence actions were received
+    let signed_tx = signed_tx_from_request(&requests[0]);
+    let actions = signed_tx.actions();
+
+    assert_eq!(
+        actions.len(),
+        expected_rollup_data_submissions.len(),
+        "received more than one action, one was supposed to fill the bundle"
+    );
+
+    for (action, expected_rollup_data_submission) in
+        actions.iter().zip(expected_rollup_data_submissions.iter())
+    {
+        let rollup_data_submission = action.as_rollup_data_submission().unwrap();
+        assert_eq!(
+            rollup_data_submission.rollup_id, expected_rollup_data_submission.rollup_id,
+            "chain id does not match. actual {:?} expected {:?}",
+            rollup_data_submission.rollup_id, expected_rollup_data_submission.rollup_id
+        );
+        assert_eq!(
+            rollup_data_submission.data, expected_rollup_data_submission.data,
+            "data does not match expected data for action with rollup_id {:?}",
+            rollup_data_submission.rollup_id,
+        );
+    }
+}
+
+/// Test to check that the executor sends a signed transaction with two sequence actions to the
+/// sequencer.
+#[tokio::test]
+async fn two_rollup_data_submissions_single_bundle() {
+    let test_composer = spawn_composer(&["test1"], None, true).await;
+    let mut composer_client = GrpcCollectorServiceClient::connect(format!(
+        "http://{}",
+        test_composer.grpc_collector_addr
+    ))
+    .await
+    .unwrap();
+
+    let response_guard =
+        mount_broadcast_tx_sync_rollup_data_submissions_mock(&test_composer.sequencer).await;
+
+    // send two sequence actions to the executor, both small enough to fit in a single bundle
+    // without filling it
+    let seq0 = RollupDataSubmission {
+        rollup_id: RollupId::new([0; ROLLUP_ID_LEN]),
+        data: vec![0u8; 1000].into(),
+        fee_asset: "nria".parse().unwrap(),
+    };
+
+    let seq1 = RollupDataSubmission {
+        rollup_id: RollupId::new([1; ROLLUP_ID_LEN]),
+        data: vec![1u8; 1000].into(),
+        fee_asset: "nria".parse().unwrap(),
+    };
+
+    // make sure at least one block has passed so that the executor will submit the bundle
+    // despite it not being full
+    time::pause();
+    time::timeout(Duration::from_millis(1000), async {
+        composer_client
+            .submit_rollup_transaction(SubmitRollupTransactionRequest {
+                rollup_id: Some(seq0.rollup_id.into_raw()),
+                data: seq0.data.clone(),
+            })
+            .await
+            .expect("rollup transactions should have been submitted successfully to grpc collector")
+    })
+    .await
+    .unwrap();
+    time::timeout(Duration::from_millis(1000), async {
```

Comment:
This sometimes times out, even when increasing the timeout time. Unsure as to if this is a problem with the test or something flaky about composer.